File: frontend/src/types/diff.ts
export enum ChangeType {
  ADDED = "added",
  REMOVED = "removed",
  MODIFIED = "modified"
}

export interface ContentChange {
  change_type: ChangeType;
  old_content: string;
  new_content: string;
  line_range_old: [number, number];
  line_range_new: [number, number];
}

export interface DiffRequest {
  old_version_id: string;
  new_version_id: string;
}

export interface DiffResponse {
  page_id: string;
  old_version_id: string;
  new_version_id: string;
  old_timestamp: string;
  new_timestamp: string;
  changes: ContentChange[];
  total_changes: number;
}

File: backend/app/schemas/diff.py
from pydantic import BaseModel
from typing import List, Optional, Tuple
from datetime import datetime
from enum import Enum

class ChangeType(str, Enum):
    ADDED = "added"
    REMOVED = "removed"
    MODIFIED = "modified"

class ContentChange(BaseModel):
    change_type: ChangeType
    old_content: str
    new_content: str
    line_range_old: Tuple[int, int]
    line_range_new: Tuple[int, int]

class DiffRequest(BaseModel):
    old_version_id: str
    new_version_id: str

class DiffResponse(BaseModel):
    page_id: str
    old_version_id: str
    new_version_id: str
    old_timestamp: datetime
    new_timestamp: datetime
    changes: List[ContentChange]
    total_changes: int